ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/connections/netty/HandshakeHandlerOutbound.java

```diff
@@ -53,7 +54,8 @@ public HandshakeHandlerOutbound(
         connectionFuture,
         connectionEventDispatcher,
         metricsSystem);
-    handshaker.prepareInitiator(kp, SECP256K1.PublicKey.create(peer.getId()));
+    handshaker.prepareInitiator(
```

Comment:
Does it make sense to move the second parameter `SECP256K1.PublicKey.create(peer.getId())` as potentially a static method in NodeKey as well?

Reply:
Thinking not - NodeKey is all about protecting this node's identity/private key - the Peer in this question is someone remote.
I.e. only operations for _our_ node key need to go via the NodeKey interface